### ISSUE TITLE
[uba] Fix new false negative from #10034

### ISSUE THREAD

**Issue by jacobtylerwalls** [pylint-dev/pylint#10069]
```
def nonlocal_in_outer_frame_ok(callback, condition_a, condition_b):
    def outer():
        nonlocal callback
        def inner():
            if condition_a:
                def inner2():
                    callback()  # possibly-used-before-assignment?
                inner2()
            else:
                if condition_b:
                    def callback():
                        pass
        inner()
    outer()
```
we should probably raise a message here?

_Originally posted by @zenlyj in https://github.com/pylint-dev/pylint/pull/10034#issuecomment-2453421927_
